### ISSUE TITLE
CallStatusFilter Label Mismatch

### ISSUE THREAD

**Issue by Scott-James-Hurley** [UserOfficeProject/user-office-project-issue-tracker#1594]
### What is the problem and why is it a problem

In `apps/frontend/src/components/call/CallStatusFilter.tsx`, the input label `call-status-select-label` is not associated with the input `call-status-select`. This means that the purpose of the input field may not be read out by screen readers. The solution should also be documented for future reference.

<img width="1723" height="797" alt="Image" src="https://github.com/user-attachments/assets/1407cc65-d2b8-4bcb-bd63-bc132e0a84b4" />
